opentracing-api/src/main/java/io/opentracing/ScopeManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+/**
+ * The {@link ScopeManager} interface abstracts both the activation of {@link Span} instances (via
+ * {@link ScopeManager#activate(Span)}) and access to an active {@link Span}/{@link Scope}
+ * (via {@link ScopeManager#active()}).
+ *
+ * @see Scope
+ * @see Tracer#setScopeManager(ScopeManager)
+ */
+public interface ScopeManager {
+    /**
+     * Make a {@link Span} instance active.
+     *
+     * @param span the {@link Span} that should become the {@link #active()}
+     * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
+     * programming error to neglect to call {@link Scope#close()} on the returned instance.
+     */
+    Scope activate(Span span);
```

Comment:
I know we've gone back and forth on this, but I think (?) I'm in favor of a varargs Observer param if we're going to have an Observer (rather than enum / bool) at all. If nothing else, it'd allow us to have only one method signature here.

Reply:
I dunno.
- What does it mean to pass multiple observers to activate? That's more machinery for implementations to agree on (or get wrong). Is it really simpler in that case?
- Can't observers be chained in various ways with a chain-able observer, thus varags are unnecessary burden for tracers? 
- FINISH_ON_CLOSE as the default is the confusing part of this API. IMHO varargs makes that harder to explain, rather than a separate method signature with corresponding separate behavior.

Reply:
Personally, I would prefer a boolean for FINISH_ON_CLOSE. I'm not a fan of the alternatives because:

a) Adding an new Enum value would be a breaking change and realistically it will never happen past the RC stage. We either can instrument things cleanly with just a boolean, or we can't. If we can't, but options here fix some issues in some instrumentations, then this is an important flex point, and we should use an Observer so that we can extend this flex point without breaking backwards compatibility. 

b) Observer: I'm no longer convinced that an Observer at this juncture would help. If we are going with the "bare bones API" approach in this RC to see if simplicity can really work, let's not be speculative and only put in what we KNOW we need. I'm willing to have some pain and an RC2 that solves that pain.